os/arch/arm/src/amebad/amebad_irq.c

```diff
@@ -349,13 +349,14 @@ void up_irqinitialize(void)
 #endif
   /* Attach all other processor exceptions (except reset and sys tick) */
 
-#ifdef CONFIG_DEBUG
-  irq_attach(AMEBAD_IRQ_NMI, amebad_nmi, NULL);
 #ifdef CONFIG_ARMV8M_MPU
   irq_attach(AMEBAD_IRQ_MEMFAULT, up_memfault, NULL);
 #endif
   irq_attach(AMEBAD_IRQ_BUSFAULT, amebad_busfault, NULL);
   irq_attach(AMEBAD_IRQ_USAGEFAULT, amebad_usagefault, NULL);
+
+#ifdef CONFIG_DEBUG
+  irq_attach(AMEBAD_IRQ_NMI, amebad_nmi, NULL);

```

Comment:
Could you let me know the criteria(?)? Why memfault, busfault and usagefault are out of CONFIG_DEBUG? Why are others in it?

Reply:
Only in these 3 cases, we would like to print additional information about the fault before calling panic. In other cases, we have nothing to print.

Reply:
@sunghan-chang 
We plan to move these fault handlers out of chip specific file into arch specific file. 
And we will print detailed information about each fault based on the fault status registers. 
There are also some unwanted code in the memfault and other fault handlers, and we will do code clean up for these.
This will be done in a new PR.

Reply:
So, can I merge current change?

Reply:
Yes. Pls merge it.

Reply:
Because of conflicts, I can't now.